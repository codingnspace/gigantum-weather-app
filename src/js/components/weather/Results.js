//Vendor
import React, {Component} from 'react';
//Locals
import bike from 'images/bicycle.svg';
import metro from 'images/metro.svg';

class Results extends Component {
  render () {
    const resultList = this.props.results && this.props.results.map((result, idx) => {
      return (
        <div key={result.idx} className="result">
          <div className='result-day-text'>{idx === 0 ? 'Today' : result.day}</div>
          <div className='result-img'>
            <img className="app-logzzo" src={result.decision === 'bike' ? bike : metro} alt="app-logo" />
          </div>
        </div>
      );
    });
    return (
      <div className="result-list">
        <h2>Your Results Are In...</h2>
        {resultList}
        <div className="start-over-btn">
          <button onClick={this.props.resetResults}>Start Over</button>
        </div>
      </div>
    );
  }
}

export default Results;
