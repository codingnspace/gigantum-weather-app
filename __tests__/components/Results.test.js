import { shallowToJson } from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import React from 'react';

import Results from 'js/components/weather/Results';
import Adapter from 'enzyme-adapter-react-15';

configure({adapter: new Adapter()});

describe('Results', () => {

  it('should render correctly', () => {
    const results = [{
      idx: 0,
      day: 'Monday',
      decision: 'metro'
    }];
    const output = shallow(
      <Results results={results} resetResults={() => null} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
