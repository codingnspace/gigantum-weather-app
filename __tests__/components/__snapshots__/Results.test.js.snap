// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render correctly 1`] = `
<div
  className="result-list"
>
  <h2>
    Your Results Are In...
  </h2>
  <div
    className="result"
    key="0"
  >
    <div
      className="result-day-text"
    >
      Today
    </div>
    <div
      className="result-img"
    >
      <img
        alt="app-logo"
        className="app-logzzo"
        src={Object {}}
      />
    </div>
  </div>
  <div
    className="start-over-btn"
  >
    <button
      onClick={[Function]}
    >
      Start Over
    </button>
  </div>
</div>
`;
